runtime/scripts/math.js

```diff
@@ -2051,6 +2051,124 @@ var math = Numbas.math = /** @lends Numbas.math */ {
             }
         }
         return factors;
+    },
+           /** Combines two matrices horizontally
+     * 
+     * @param {matrix} m1
+     * @param {matrix} m2
+     * @returns {matrix}
+     */
+            combine_horizontally: function(m1,m2){
+                m1.columns = m1[0].length;
+                m1.rows = m1.length;
+        
+                m2.columns = m2[0].length;
+                m2.rows = m2.length;
+        
+                for(var i = 0; i < m2.columns+1; i++) {
+        
+                    if (m2.rows > m1.rows){
+                        for(var j=m1.rows;j<m2.rows;j++){
+                        m1[m1.rows]=[]
+                            for (var x = 0; x < m1.columns; x++){
+                                m1[m1.rows].push('0');
+                            }
+                        m1.rows++
+                    }
+                    }
+                    else if (m2.rows < m1.rows){
+                        for(var j=m2.rows;j<m1.rows;j++){
+                            m2[m2.rows]=[]
+                        for (var y = 0; y < m2.columns; y++){
+                            m2[m2.rows].push('0');
+                        }
+                        m2.rows++
+                    }
+                    }
+                }
+                var m3=m1.concat(m2)
```

Comment:
This will put all the rows of `m2` under the rows of `m1`. Is that what you want when combining horizontally?

It looks like you end up with `m1` and `m2` both having the same number of rows, but not necessarily the same number of columns, and maybe you think that `.concat` will concatenate each pair of rows?